example/memcached-operator/memcached_controller.go.tmpl

```diff
@@ -85,7 +87,7 @@ type ReconcileMemcached struct {
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	log.Printf("Reconciling Memcached %s/%s\n", request.Namespace, request.Name)
+	log.Info("Reconciling Memcached", "Request.Namespace", request.Namespace, "Requst.Name", request.Name)
```

Comment:
Seeing a log message formatted like this is quite weird after using Printf, Logf etc for so long.
@estroz Can we format the string with `fmt.Sprintf()`? That may be a bit verbose though.
```Go
log.Info(fmt.Sprintf("Reconciling Memcached %s/%s\n", request.Namespace, request.Name))
```
Seems like [logr.Info](https://github.com/go-logr/logr/blob/master/logr.go#L107-L111) just accepts keys/values.

@shawn-hurley This format of constructing log messages seems a bit cumbersome to me at least. Not sure how users would feel if this is the default.

Reply:
I agree that this interface pushes users away from unstructured string formatting (and toward structured logging). 

It lends itself to building a scoped logger with `WithName` and/or `WithValues` so that users don't have to repeat the same string formatting for variables every time they need to log a message.
```go
requestLogger := log.WithFields("Request.Namespace", request.Namespace, "Request.Name", request.Name)
requestLogger.Info("Reconciling Memcached")
...
requestLogger.Info("Memcached resource not found. Ignoring since object must be deleted")
```

My two cents is that the nudge toward structured logging is probably a good thing since the interface makes it cleaner to log simple messages that share common metadata, and it makes logs more machine-consumable by default, which is probably a good habit to reinforce.

Reply:
I +1 basically everything @joelanford said above.  As stated earlier in this conversation as well, if users of the SDK hate it they can choose a different logger. I think to have this as a default but maybe (later) adding a doc showing how to go back to logrus will show users they have a choice. I think in that doc it would be smart though to explain what Joe said above. 

Thoughts?

Reply:
Yeah I agree. It's actually good if we're pushing users towards structured logging as the default.
I guess my initial reaction was prompted by being used to unstructured logging myself.

And yeah a doc to illustrate the choice of using a different logger should be good enough if users want to opt out of the default.

Reply:
@hasbro17 @shawn-hurley @joelanford added a doc on logging.